tools/partutil/extend_partition.go

```diff
@@ -0,0 +1,181 @@
+	tableByte, err := exec.Command("sudo", "sfdisk", "--dump", disk).Output()
+	if err != nil {
+		return fmt.Errorf("error in dumping partition table of %s, "+
+			"input: disk=%s, partNumInt=%d, end sector=%d, "+
+			"error msg: (%v)", disk, disk, partNumInt, end, err)
+	}
+
+	// edit partition table.
+	tableString, err := editPartitionEnd(string(tableByte), partName, end)
+	if err != nil {
+		return fmt.Errorf("error when editing partition table of %s to %d, "+
+			"input: disk=%s, partNumInt=%d, end sector=%d, "+
+			"error msg: (%v)", disk, end, disk, partNumInt, end, err)
+	}
+
+	tableBuffer.WriteString(tableString)
+
+	// write partition table back.
+	writeTableCmd := exec.Command("sudo", "sfdisk", "--no-reread", disk)
+	writeTableCmd.Stdin = &tableBuffer
+	writeTableCmd.Stdout = os.Stdout
+	if err := writeTableCmd.Run(); err != nil {
+		return fmt.Errorf("error in writing partition table back to %s, "+
+			"input: disk=%s, partNumInt=%d, end sector=%d, "+
+			"error msg: (%v)", disk, disk, partNumInt, end, err)
+	}
+
+	log.Printf("\nCompleted extending %s\n\n", partName)
+
+	// check and repair file system in the partition.
+	cmd := fmt.Sprintf("sudo e2fsck -fp %s", partName)
+	if err := ExecCmdToStdout(cmd); err != nil {
+		return fmt.Errorf("error in checking file system of %s, "+
+			"input: disk=%s, partNumInt=%d, end sector=%d, "+
+			"error msg: (%v)", partName, disk, partNumInt, end, err)
+	}
+	log.Printf("\nCompleted checking file system of %s\n\n", partName)
+
+	// resize file system in the partition.
+	cmd = fmt.Sprintf("sudo resize2fs %s", partName)
+	if err := ExecCmdToStdout(cmd); err != nil {
+		return fmt.Errorf("error in resizing file system of %s, "+
+			"input: disk=%s, partNumInt=%d, end sector=%d, "+
+			"error msg: (%v)", partName, disk, partNumInt, end, err)
+	}
+
+	log.Printf("\nCompleted updating file system of %s\n\n", partName)
+	return nil
+}
+
+// When we read disk information by dumping the partition table, we get output like the following:
+// sudo sfdisk --dump /dev/sdb
+// label: gpt
+// label-id: 8071096F-DA33-154D-A687-AE097B8252C5
+// device: /dev/sdb
+// unit: sectors
+// first-lba: 2048
+// last-lba: 20971486
+
+// /dev/sdb1 : start=     4401152, size=     2097152, type=0FC63DAF-8483-4772-8E79-3D69D8477DE4, uuid=3B41256B-E064-544A-9101-D2647C0B3A38
+// /dev/sdb2 : start=      206848, size=     4194304, type=0FC63DAF-8483-4772-8E79-3D69D8477DE4, uuid=60E55EA1-4EEA-9F44-A066-4720F0129089
+// /dev/sdb3 : start=     6498304, size=      204800, type=0FC63DAF-8483-4772-8E79-3D69D8477DE4, uuid=9479C34A-49A6-9442-A56F-956396DFAC20
+
+// editPartitionEnd changes partition end in the partition table string to extend the partition.
+func editPartitionEnd(table, partName string, end int) (string, error) {
+	var err error
+	lines := strings.Split(table, "\n")
+	haveValidPartition := false // whether has the required partition.
+	for i, line := range lines {
+		if strings.Contains(line, partName) {
```

Comment:
It's generally better to keep the main logic unindented, if possible. Consider writing this the following way instead:

if !strings.Contains(line, partName) {
  continue
}
ls := strings.Split(...)